web/src/recent_view_ui.js

```diff
@@ -911,17 +943,23 @@ export function complete_rerender() {
     });
 }
 
-export function show() {
+export function show(for_stream_id = null) {
     views_util.show({
-        highlight_view_in_left_sidebar: left_sidebar_navigation_area.highlight_recent_view,
+        highlight_view_in_left_sidebar() {
+            if (for_stream_id === null) {
+                left_sidebar_navigation_area.highlight_recent_view();
+            }
```

Comment:
Should this have an `else` clause to highlight the current stream in the left sidebar? Might make sense to leave for a follow-up PR and just have a comment here, since I'm not sure what the semantics should be.

Reply:
Yeah I'm also not sure what we want to do there. I can leave a comment.

Reply:
Oh I was actually already doing this elsewhere -- highlighting the stream looks a lot better than keeping the recent conversations highlighted, in my opinion. I'll put that code here instead.

I'm making a separate commit to show this change since I'm still not really confident if code is in the ideal place and want to my progress. I'd love some feedback on it.

Reply:
Yeah agree from a product perspective, we should be highlighting the stream (and probably expanding it in the sidebar), if we're trying to hook it in as something you access via the left sidebar.

If we want to instead hook it in via, say, a stream picker component in the Recent view, that might not make sense.